### DISCUSSION TITLE
[QUESTION] How to get transaction fees from a getblock hex-encoded data?

### DISCUSSION THREAD

**Discussion by ababo** [rust-bitcoin/rust-bitcoin#1767] [Q&A]
When I parse the hex-encoded data received using "getblock" RPC call (with verbosity 0) I get a `Block` instance. The problem that it contains `Transaction` instances rather than `PartiallySignedTransaction` instances, so I can't retrieve fees. Is there a way to compute fee for each of the `Transaction` instance above?

**Comment by sanket1729** [comment]
For getting the fees, you need the previous spent output for that transaction which the `getblock` hex does not have. The input amounts in transactions are implicitly calculated as previous spent output which is most likely in another block.

**Comment by ababo** [comment]
Can I get the fee using "gettransaction"?

**Comment by ababo** [comment]
I would rephrase my question. Is there a way to get tx fee using the RPC API?

**Comment by sanket1729** [comment]
I don't know of a direct way to get the fees a transaction via `bitcoind` RPC. That said, I am not a heavy RPC user of it. But I am sure, if there is an easy way, your question can be better answered at https://bitcoin.stackexchange.com/ .

A naive way to doing it would require getting all output amounts from the current transaction, and calling `gettxout <txid> <vout>` for each input to get the corresponding spending input and recalcuate the sum from there.

**Comment by Kixunil** [comment]
Moving to discussion since this is not Rust-related.
